tests/test_aws.py

```diff
@@ -142,6 +142,15 @@ def test_create_instances_missing_instance_type(aws):
         aws.create_instances()
 
 
+def test_create_instances_missing_region(aws):
+    aws.region_name = None
+    os.environ.pop("AWS_DEFAULT_REGION")
```

Comment:
You might do something here to replace the existing version at the end of the test. This doesn't affect the outside environment, but it could affect other tests if they expect to see this set? (But I guess the tests shouldn't be expecting this to be set, so it probably doesn't matter.)

Reply:
We set the environment variables on each run as part of the `aws` fixture.

https://github.com/omsf-eco-infra/gha-runner/blob/a7cd6250081b92e92f1107da354b34e425d2f8e0/tests/test_aws.py#L15

So I am not sure this is necessary.

Reply:
Good point! That should be fine then.